chips/nrf52840/src/ieee802154_radio.rs

```diff
@@ -1298,24 +1349,24 @@ impl<'a> kernel::hil::radio::RadioData<'a> for Radio<'a> {
         buf: &'static mut [u8],
         frame_len: usize,
     ) -> Result<(), (ErrorCode, &'static mut [u8])> {
-        if self.tx_buf.is_some() {
-            // tx_buf TakeCell is only occupied when a transmission is underway. This
-            // check insures we do not interrupt an ungoing transmission
+        if self.state.get() == RadioState::OFF {
+            return Err((ErrorCode::OFF, buf));
+        } else if self.tx_buf.is_some() {
+            // tx_buf is only occupied when a transmission is underway. This
+            // check insures we do not interrupt an ongoing transmission.
             return Err((ErrorCode::BUSY, buf));
-        } else if radio::PSDU_OFFSET + frame_len >= buf.len() {
+        } else if buf.len() < radio::PSDU_OFFSET + frame_len + radio::MFR_SIZE {
             // Not enough room for CRC
```

Comment:
```suggestion
            // Provided buffer does not have adequate space.
```

Reply:
This seems less clear to me.

Reply:
Only mentioning the CRC here is inaccurate since it may be that there is not enough room for the PHR or that there is not enough room for the CRC. Perhaps updating then to:
`// Not enough room for CRC and/or PHR.`